fluentgen/src/main/java/com/azure/autorest/fluent/mapper/ExampleParser.java

```diff
@@ -262,88 +268,104 @@ private static FluentClientMethodExample parseMethodForExample(MethodGroupClient
-                                ModelProperty modelProperty = stage.getModelProperty();
-                                if (modelProperty != null) {
-                                    ExampleNode node = parseNodeFromModelProperty(example, requestBodyParameter, requestBodyClientModel, modelProperty);
-
-                                    if (stage.isMandatoryStage() || !node.isNull()) {
-                                        exampleNodes.add(node);
-                                    }
-                                }
-                            }
-
-                            if (exampleNodes.stream().anyMatch(ExampleNode::isNull)) {
-                                if (stage.isMandatoryStage()) {
-                                    logger.warn("Failed to assign sample value to required stage '{}'", stage.getName());
-                                }
-                            }
-
-                            if (!exampleNodes.isEmpty()) {
-                                resourceCreateExample.getParameters().add(new ParameterExample(fluentMethod, exampleNodes));
-                            }
+                } else {
+                    ModelProperty modelProperty = stage.getModelProperty();
+                    if (modelProperty != null) {
+                        ExampleNode node = parseNodeFromModelProperty(example, requestBodyParameter, resourceCreate.getRequestBodyParameterModel(), modelProperty);
+
+                        if (stage.isMandatoryStage() || !node.isNull()) {
+                            exampleNodes.add(node);
                         }
                     }
+                }
 
-                    ret.add(resourceCreateExample);
+                if (exampleNodes.stream().anyMatch(ExampleNode::isNull)) {
+                    if (stage.isMandatoryStage()) {
+                        logger.warn("Failed to assign sample value to required stage '{}'", stage.getName());
+                    }
+                }
+
+                if (!exampleNodes.isEmpty()) {
+                    resourceCreateExample.getParameters().add(new ParameterExample(fluentMethod, exampleNodes));
                 }
             }
         }
-        return ret;
+        return resourceCreateExample;
+    }
+
+    public static FluentResourceCreateExample parseResourceCreate(FluentResourceCollection resourceCollection, ResourceCreate create, ProxyMethodExample example) {
+        List<MethodParameter> methodParameters = getParameters(
+            create.getMethodReferences()
+                .stream()
+                .filter(collectionMethod-> requiresExample(collectionMethod.getInnerClientMethod()))
+                .findFirst().get()
+                .getInnerClientMethod());
+        MethodParameter requestBodyParameter = methodParameters.stream()
+            .filter(p -> p.getProxyMethodParameter().getRequestParameterLocation() == RequestParameterLocation.BODY)
+            .findFirst().orElse(null);
```

Comment:
nit, this line happens in the class for 4 times, better to move to a method.

Reply:
Extracted a new method `findRequestBodyParameter`.